ddtrace/propagation/http.py

```diff
@@ -1152,6 +1152,15 @@ def my_controller(url, headers):
                         context._baggage = baggage_context.get_all_baggage_items()
                     else:
                         context = baggage_context
+                    if config._baggage_tag_keys:
+                        allowed_keys = {k.strip() for k in config._baggage_tag_keys if k.strip()}
+                        for key, value in baggage_context.get_all_baggage_items().items():
+                            if key in allowed_keys:
+                                prefixed_key = "baggage." + key
+                                if prefixed_key not in context._meta:
+                                    context._meta[prefixed_key] = value
+                                else:
+                                    log.debug("Skipping baggage tag '%s'; tag already set", prefixed_key)
```

Comment:
Would it be more performant to only get the baggages we need, rather than looping through all of them each time?

```suggestion
                        for baggage_tag_key in config._baggage_tag_keys:
                            if (stripped_key := baggage_tag_key.strip()):
                                span_tag_key = "baggage." + stripped_key
                                if span_tag_key not in context._meta:
                                    context._meta[prefixed_key] = baggage_context.get_baggage(stripped_key)
                                else:
                                    log.debug("Skipping baggage tag '%s'; tag already set", prefixed_key)
```

Reply:
Made the changes to only loop through the keys that are specified via the configuration.